include/usb/class/usb_hid.h

```diff
@@ -57,11 +57,12 @@ struct usb_hid_descriptor {
 
 /* Public headers */
 
-typedef int (*hid_cb_t)(struct usb_setup_packet *setup, int32_t *len,
+typedef int (*hid_cb_t)(const struct device *dev,
+			struct usb_setup_packet *setup, int32_t *len,
 			uint8_t **data);
-typedef void (*hid_int_ready_callback)(void);
-typedef void (*hid_protocol_cb_t)(uint8_t protocol);
-typedef void (*hid_idle_cb_t)(uint16_t report_id);
+typedef void (*hid_int_ready_callback)(const struct device *dev);
```

Comment:
The commit introduces incompatible changes and has to go through the "stable API change" process. You might consider backwards compatibility, but it probably will not be that nice.

Reply:
I can post an Issue with stable API change, the question is do You agree that this is change worth pursuing?

Reply:
>  the question is do You agree that this is change worth pursuing?

From the commit message, it seems to be reasonable.